### PR TITLE
Optimizations

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: libethash/CMakeLists.txt [hunk 1]
```diff
@@ -29,13 +29,7 @@ if (NOT CRYPTOPP_FOUND)
 	find_package(CryptoPP 5.6.2)
 endif()
 
-if (CRYPTOPP_FOUND)
```
Comment (bobsummerwill) [inline libethash/CMakeLists.txt]: Great!

See also https://github.com/ethereum/cpp-ethereum/issues/3088.   If that is something which you would like to look at, that would be fantastic.

File: libdevcore/CommonData.cpp [hunk 2]
```diff
@@ -35,13 +35,26 @@
...
 
-bool dev::isHex(string const& _s)
+namespace
 {
-	unsigned i = (_s.size() >= 2 && _s.substr(0, 2) == "0x") ? 2 : 0;
-	for (; i < _s.size(); ++i)
-		if (fromHex(_s[i], WhenError::DontThrow) == -1)
-			return false;
-	return true;
+int _fromHex(char _i) noexcept
```
Comment (chfast) [inline libdevcore/CommonData.cpp]: It conflicts with `fromHex(string)`. I didn't have better name.

Comment (chriseth) [inline libdevcore/CommonData.cpp]: We could name it `fromHexNibble` / `fromHexChar` but we can also leave it like that.

Comment (bobsummerwill) [inline libdevcore/CommonData.cpp]: Let's change the name.    The underscore will have everybody who reads the code asking what the underscore means.

Better to just have a more explicit, longer name, which describes what exactly the function is.

**Comment by chfast** [comment]
Updated. Please review.

**Pull request merged**